clvm/SExp.py

```diff
@@ -16,27 +17,27 @@
 
 CastableType = typing.Union[
     "SExp",
-    "CLVMObject",
+    CLVMStorage,
     bytes,
     str,
     int,
     None,
-    list,
-    typing.Tuple[typing.Any, typing.Any],
+    typing.Sequence["CastableType"],
+    typing.Tuple["CastableType", "CastableType"],
 ]
 
 
 NULL = b""
 
 
-def looks_like_clvm_object(o: typing.Any) -> bool:
+def looks_like_clvm_object(o: typing.Any) -> typing_extensions.TypeGuard[CLVMStorage]:
     d = dir(o)
     return "atom" in d and "pair" in d
 
 
 # this function recognizes some common types and turns them into plain bytes,
 def convert_atom_to_bytes(
-    v: typing.Union[bytes, str, int, None, list],
+    v: typing.Union[bytes, str, int, None, typing.List[typing_extensions.Never], typing.SupportsBytes],
```

Comment:
couldn't this just be typed as `CastableType`?

Reply:
For one, this only accepts an empty list.  So not by trivially replacing the hint at least, no.

Reply:
oh, right. I was thinking of `SExp.to()`